src/internal/transactionenv/env.go

```diff
@@ -67,6 +67,7 @@ type TransactionServer interface {
 
 type PFSBackend interface {
 	NewPropagater(*txncontext.TransactionContext) txncontext.PfsPropagater
+	NewRepoChecker(*txncontext.TransactionContext) txncontext.PfsBranchChecker
```

Comment:
Maybe this should be called `RepoValidator`, and the method on the transaction context would be `ValidateRepo`.